server/src/test/java/io/druid/metadata/TestDerbyConnector.java

```diff
@@ -18,31 +18,112 @@
 package io.druid.metadata;
 
 import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import io.druid.metadata.storage.derby.DerbyConnector;
 import org.junit.Assert;
+import org.junit.rules.ExternalResource;
 import org.skife.jdbi.v2.DBI;
 import org.skife.jdbi.v2.exceptions.UnableToObtainConnectionException;
 
 import java.sql.SQLException;
+import java.util.UUID;
 
 public class TestDerbyConnector extends DerbyConnector
 {
+  public static class DerbyConnectorRule extends ExternalResource
+  {
+    private TestDerbyConnector connector;
+    private final Supplier<MetadataStorageTablesConfig> dbTables;
+    private final MetadataStorageConnectorConfig connectorConfig;
+
+    public DerbyConnectorRule()
+    {
+      this(Suppliers.ofInstance(MetadataStorageTablesConfig.fromBase("druidTest" + dbSafeUUID())));
```

Comment:
I think it's good to be able to test table creation both with default names and with a specified prefix, so let's remove it if it's not necessary. Each test method already gets it's unique database, so that should be enough. It will make the test code simpler and less confusing.